hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -69,6 +74,7 @@ public HiveCatalog(Configuration conf) {
     this.conf = conf;
     this.createStack = Thread.currentThread().getStackTrace();
     this.closed = false;
+    this.defaultFileIO = new HadoopFileIO(conf);
```

Comment:
Instantiating `HadoopFileIO` before `initialize` is called is because it isn't always called right now?